src/frontend/models.ts

```diff
@@ -0,0 +1,9 @@
+export interface IUseCase {
```

Comment:
I think that model.ts and state.ts can be unified. No need to have two files, they basically do the same thing. I don't care how you call it though, so maybe move state.ts into models.ts....

Reply:
Ok, sounds good, I would prefer to call it models if that's ok with you. I've seen "interfaces" also but I prefer models.